admin_api.go

```diff
@@ -79,6 +79,24 @@ func (a *AdminAPI) XPubs(ctx context.Context, opts ...queries.QueryOption[filter
 	return res, nil
 }
 
+// CreateContact creates a new contact record via the Admin Contacts API.
+// It accepts a command containing the necessary parameters to define the contact record,
+// such as the creator's paymail, contact's full name, paymail and any associated metadata.
+//
+// The method sends a request to the Contacts API to create the contact, and the API
+// response is unmarshaled into a *response.Contact struct, representing the newly created contact.
+//
+// If the API request fails or the response cannot be decoded, an error is returned. The error is
+// wrapped with additional context to assist in troubleshooting.
+func (a *AdminAPI) CreateContact(ctx context.Context, cmd *commands.CreateContact) (*response.Contact, error) {
+	res, err := a.contactsAPI.CreateContact(ctx, cmd)
+	if err != nil {
+		return nil, contacts.HTTPErrorFormatter("create user contacts page", err).FormatPostErr()
```

Comment:
@mgosek-4chain why some packages have this method `HTTPErrorFormatter` and there is package called `errutil` which contain exactly the same thing 

```go
type HTTPErrorFormatter struct {
	Action string
	API    string
	Err    error
}

func (h HTTPErrorFormatter) Format(method string) error {
	return fmt.Errorf("failed to send HTTP %s request to %s via %s: %w", method, h.Action, h.API, h.Err)
}

func (h HTTPErrorFormatter) FormatPutErr() error    { return h.Format(http.MethodPut) }
func (h HTTPErrorFormatter) FormatPatchErr() error  { return h.Format(http.MethodPatch) }
func (h HTTPErrorFormatter) FormatPostErr() error   { return h.Format(http.MethodPost) }
func (h HTTPErrorFormatter) FormatGetErr() error    { return h.Format(http.MethodGet) }
func (h HTTPErrorFormatter) FormatDeleteErr() error { return h.Format(http.MethodDelete) }
```

```
internal/api/v1/admin/utxos/utxos_api.go:55:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/admin/contacts/contacts_api.go:113:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/admin/paymails/paymails_api.go:103:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/admin/status/status_api.go:42:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/admin/transactions/transactions_api.go:68:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/admin/accesskeys/access_keys_api.go:55:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/admin/xpubs/xpubs_api.go:76:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/admin/webhooks/webhooks_api.go:53:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/admin/stats/stats_api.go:41:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/admin/invitations/invitations_api.go:52:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/user/utxos/utxos_api.go:61:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/user/contacts/contacts_api.go:137:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/user/merkleroots/merkleroots_api.go:68:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/user/paymails/paymails_api.go:61:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/user/transactions/transactions_api.go:186:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/user/accesskeys/access_key_api.go:103:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/user/configs/configs_api.go:44:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/user/xpubs/xpub_api.go:59:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
internal/api/v1/user/invitations/invitations_api.go:53:func HTTPErrorFormatter(action string, err error) *errutil.HTTPErrorFormatter {
```

do we need so many methods like this ? would not be better to have one under one package ?

Reply:
I completely agree.  In the initial iteration of "v2", we aimed to avoid premature optimizations, including introducing generics. However, as you can see (for ex. _queries pkg_), we’ve started improving the codebase by reducing duplications. This will definitely be a candidate to address after the release.

Reply:
The task for the codebase standardization has been added and assigned to you :)